### DISCUSSION TITLE
nz-table scss problem

### DISCUSSION THREAD

**Discussion by Berius** [NG-ZORRO/ng-zorro-antd#8565] [Q&A] [answered]
In our project we have problem with nz-table component. It doesn't look like the component on ng-zorro page.
Expected look:
<img width="1034" alt="image" src="https://github.com/NG-ZORRO/ng-zorro-antd/assets/26148651/ec2a9bad-3d7e-40a5-80f3-2cd2efdd4305">
In our project:
<img width="1227" alt="image" src="https://github.com/NG-ZORRO/ng-zorro-antd/assets/26148651/a16a7d4a-a807-4c4c-8d36-6645d5ee6428">
We made a new project to check this table, but result was the same.
Thank you in advance.

**Comment by crabvendor** [comment, marked as answer]
Having similar issues, on top of this:
- table cannot be styled by overriding some less variables; eg. table font size works fine, but hover color doesn't
- style files are imported correctly, other imported components works fine
- empty indicator doesn't show up when table is empty
```LESS

// Custom Theming for NG-ZORRO
// For more information: https://ng.ant.design/docs/customize-theme/en
@import "../node_modules/ng-zorro-antd/ng-zorro-antd.less";

// @primary-color: #1890ff;
@table-header-bg: red;
@table-border-color: red;
@table-row-hover-bg: red;

@table-font-size: 40px;

@primary-color: green;
```

Result:
<img width="1512" alt="image" src="https://github.com/NG-ZORRO/ng-zorro-antd/assets/39218310/5e724188-f6da-44c4-8e11-7dd34f1a2c1b">

Empty table:
<img width="1394" alt="image" src="https://github.com/NG-ZORRO/ng-zorro-antd/assets/39218310/02f0c92a-c6c3-415c-a1ec-fecb570d37c6">

ng-zorro-antd version: 17.4.0
angular version: 17.3.0
node: tried both 20.9 and 20.11

Initially expected some of the project global css to interfere but creating new project from scratch doesn't work either, doesn't work on any machine we tried it on.

Noticed in console that some of the inner component classes are missing:
Expected:
<img width="338" alt="image" src="https://github.com/NG-ZORRO/ng-zorro-antd/assets/39218310/5c62b9cc-131a-41a1-a94f-d5c71576643d">

Actual:
<img width="457" alt="image" src="https://github.com/NG-ZORRO/ng-zorro-antd/assets/39218310/a9a00fee-ebae-4fb5-8715-244e60cb1923">

After manually adding classes to the components in dev tools it reassembles it bit more:
<img width="1508" alt="image" src="https://github.com/NG-ZORRO/ng-zorro-antd/assets/39218310/4e9ed29a-c770-41f8-986d-ba6a09159af5">

Problem resolved by downgrading to library version 17.0.0

Ok last edit, since I think I've found the underlying issue - downgrading to 17.0.0 only helps because it forces to create module file and import `NzTableModule` in our standalone component, so the actual version doesn't matter once module file is there.

Problem is that when using NzTable as standalone component, it most likely doesn't import it's underlying children component like `NzTableModule` does.

Problem is resolved by either importing them all as standalone components alongside `NzTableComponent` or Importing `NzTableModule` on it's own.

**Reply by Berius** [reply]
Wow, great work.
Thanks!
